text/0000-pack-pull-policy.md

```diff
@@ -0,0 +1,72 @@
+# Meta
+[meta]: #meta
+- Name: `pack` Image Pull Policy
+- Start Date: 2020-05-21
+- Author(s): @jromero
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Replace the flag `--no-pull` with `--pull-policy <option>` and default to "IfNotPresent"
+
+# Motivation
+[motivation]: #motivation
+
+#### Performance
+
+By not having to check for the latest images on every build we improve performance.
+
+#### Uncommon
+
+The docker CLI, docker compose, and k8s default to similar functionality as described by "IfNotPresent".
+
+# What it is
+[what-it-is]: #what-it-is
+
+A new flag `--pull-policy` that takes the following possible options:
+
+- `if-no-present` - only attempt to pull the images if it's not present
+- `always` - attempt to always pull images
+- `never`- don't attempt to pull images
+
+# How it Works
+[how-it-works]: #how-it-works
+
+<!-- This is the technical portion of the RFC, where you explain the design in sufficient detail.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work. -->
+
+### `pack build`
+
+// TODO: Detail all the images used during build and ensure that a single flag/policy makes sense for all images or detail exclusions.
```

Comment:
One to keep in mind is the lifecycle image. Given that this image is relatively behind the scenes, it might not make sense to have the same pull policy as builders, for example (even though this is what we do today).

Reply:
I would agree with this but given that it would add complexity and confusion. I would question why they should differ?

What are some cases in which you would specify `--no-pull` (or later `--pull-policy never`) and expect any image to be pulled. Even in the scenario of a lifecycle image I would think that if I say `never`, I mean `never` and would have all the bits necessary locally (eg. offline).

Reply:
We could always expand this later to accept a more fine-grained pull policy: e.g. `--pull-policy builder=always,lifecycle=not-present`. But I think it's okay to wait until we have validated use cases for this.

Reply:
I think `if-not-present` satisfies my concern - e.g., the pack user would like NOT to pull the builder image, but has no idea that the lifecycle image even exists and is subsequently confused when the build fails due to the lifecycle image being missing. With `if-not-present` we'll use the local builder and the lifecycle image will be quietly pulled and the user never has to care.